src/mapblock.cpp

```diff
@@ -370,7 +340,9 @@ void MapBlock::serialize(std::ostream &os_compressed, u8 version, bool disk, int
 	u8 flags = 0;
 	if(is_underground)
 		flags |= 0x01;
-	if(getDayNightDiff())
+	// This flag used to be day-night-differs,
+	// make sure it's always true for older versions
+	if(isAir() || version < 30)
```

Comment:
trying to reuse this only complicates things and we don't have shortage of bits.

Reply:
I've since changed it. It's not used anymore by new server, but old server should still be able to make use of a map even when modified by a newer server.
So new server now write !isAir() into this bit.